schemas/pulse-job.yml

```diff
@@ -0,0 +1,277 @@
+$schema: "http://json-schema.org/draft-04/schema#"
+title: "Job Definition"
+description: |
+  Definition of a single job that can be added to Treeherder
+  Project is determined by the routing key, so we don't need to specify it here.
+id: "jobDefinition"
+type: "object"
+properties:
+  jobGuid:
+    title: "jobGuid"
+    type: "string"
+    pattern: "^[A-Za-z0-9_/+-]+$"
+    minLength: 1
+    maxLength: 50
+
+  origin:
+    anyOf:
+      - type: "object"
+        properties:
+          kind:
+            type: "string"
+            enum: ['hg.mozilla.org']
+          project:
+            type: "string"
+            pattern: "^[A-Za-z0-9_-]+$"
+            minLength: 1
+            maxLength: 50
+          revision:
+            type: "string"
+            minLength: 1
+            maxLength: 50
+          pushLogID:
+            type: "integer"
+        required: [kind, project, revision]
+
+      - type: "object"
+        properties:
+          kind:
+            type: "string"
+            enum: ['github.com']
+          project:
+            type: "string"
+            pattern: "^[A-Za-z0-9_-]+$"
+            minLength: 1
+            maxLength: 50
+          revision:
+            type: "string"
+            minLength: 1
+            maxLength: 50
+          pullRequestID:
+            type: "integer"
+        required: [kind, project, revision]
+
+  display:
+    type: "object"
+    properties:
+      jobSymbol:
+        title: "jobSymbol"
+        type: "string"
+        pattern: "^[A-Za-z0-9_-]+$"
+        minLength: 1
+        maxLength: 25
+      groupSymbol:
```

Comment:
Currently we send in '?' for no group symbol...would that not be allowed anymore for this?  What should we send for group symbol if there is none (a 'B' that appears outside of a group)